src/main/java/com/bandwidth/sqs/consumer/Consumer.java

```diff
@@ -270,8 +270,9 @@ void processNextMessage() {
         }
         Message message = timedMessage.getMessage();
 
+        SqsAsyncIoClient sqsClient = manager.getSqsClient();
         MessageAcknowledger<Message> acknowledger =
-                new DefaultMessageAcknowledger(manager.getSqsClient(), queueUrl, message.getReceiptHandle());
+                new MessageAcknowledger<>(sqsClient, queueUrl, message.getReceiptHandle(), sqsClient);
```

Comment:
I knew you would say something about that. Not sure how to avoid this,
first one is passed in as a client, so it can delete/change visibility of messages. The 2nd is a `MessagePublisher` and in this case it happens to be a publisher for `Message` which client implements. (But in other cases it is usually some kind of "model" object that can be serialized)